### PR TITLE
oss: support get sts endpoint from env for private cloud

### PR DESCRIPTION
**Pull request by AlbeeSo**
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: deploy/chart/templates/controller.yaml [hunk 1]
```diff
@@ -315,6 +315,10 @@ spec:
             - name: ECS_ENDPOINT
               value: {{ .Values.deploy.ecsEndpoint | quote }}
 {{- end }}
+{{- if .Values.deploy.stsEndpoint }}
+            - name: STS_ENDPOINT
```
Comment (mowangdk) [inline deploy/chart/templates/controller.yaml:319]: STS_ENDPOINT should be renamed to RRSA_STS_ENDPOINT, as it's currently only used in RRSA

Comment (huww98) [reply deploy/chart/templates/controller.yaml:319]: But if we invoke other APIs of STS in the future, we should still use this endpoint.
BTW, RRSA is a concept for ACK, it should be OIDC for STS.

File: pkg/oss/utils.go [hunk 2]
```diff
@@ -68,6 +68,21 @@ func getRRSAConfig(opt *Options, m metadata.MetadataProvider) (rrsaCfg *mounter.
 	return &mounter.RrsaConfig{ServiceAccountName: saName, OidcProviderArn: oidcProviderArn, RoleArn: roleArn}, nil
 }
 
+// getSTSEndpoint get STS endpoint
+func getSTSEndpoint(region string) (endpoint string) {
```
Comment (huww98) [inline pkg/oss/utils.go:72]: Don't use named return value here.

File: pkg/oss/utils_test.go [hunk 3]
```diff
@@ -279,3 +280,47 @@ func Test_setTransmissionProtocol(t *testing.T) {
...
+			envEndpoint: "sts-vpc.cn-beijing.aliyuncs.com",
+			expected:    "sts-vpc.cn-beijing.aliyuncs.com",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.envEndpoint != "" {
+				os.Setenv("STS_ENDPOINT", tt.envEndpoint)
+				defer os.Unsetenv("STS_ENDPOINT")
```
Comment (huww98) [inline pkg/oss/utils_test.go:315]:
```suggestion
				t.Setenv("STS_ENDPOINT", tt.envEndpoint)
```

File: deploy/chart/templates/controller.yaml (hunk 1 above)
Comment (mowangdk) [reply deploy/chart/templates/controller.yaml:319]: We should modify the endpoint in metadata as well to match stsclient in the metadata openapi fetcher

Comment (mowangdk) [reply deploy/chart/templates/controller.yaml:319]: To utilize the universal STS_ENDPOINT  env name.

File: pkg/oss/utils.go [hunk 4]
```diff
@@ -68,6 +68,21 @@ func getRRSAConfig(opt *Options, m metadata.MetadataProvider) (rrsaCfg *mounter.
...
+
+	// for PrivateCloud
+	if os.Getenv("STS_ENDPOINT") != "" {
+		endpoint = os.Getenv("STS_ENDPOINT")
```
Comment (mowangdk) [inline pkg/oss/utils.go:76]: if we don't want to validate endpoint, how about return the endpoint directly

File: deploy/chart/templates/controller.yaml (hunk 1 above)
Comment (AlbeeSo) [reply deploy/chart/templates/controller.yaml:319]: I submit a new commit to set endpoint when try new a stsclient.

File: pkg/utils/util.go [hunk 5]
```diff
@@ -268,14 +269,18 @@ func NewEcsClient(ac AccessControl) (ecsClient *ecs.Client) {
 
 // NewStsClient create a stsClient object
 func NewStsClient(ac AccessControl) (stsClient *sts.Client) {
+	regionID, _ := GetRegionID()
```
Comment (mowangdk) [inline pkg/utils/util.go:272]: ```GetRegionID()``` should be deprecated, use default region here would be fine

**Pull request merged**